internal/buffer/save.go

```diff
@@ -92,53 +127,6 @@ func overwriteFile(name string, enc encoding.Encoding, fn func(io.Writer) error,
 		}
 	}
 
-	return
-}
-
-func (b *Buffer) overwrite(name string, isBackup bool, withSudo bool) (err error) {
-	enc, err := htmlindex.Get(b.Settings["encoding"].(string))
-	if err != nil {
-		return
-	}
-
-	var size int
-	fwriter := func(file io.Writer) (e error) {
-		b.Lock()
-		defer b.Unlock()
-
-		if len(b.lines) == 0 {
-			return
-		}
-
-		// end of line
-		var eol []byte
-		if b.Endings == FFDos {
-			eol = []byte{'\r', '\n'}
-		} else {
-			eol = []byte{'\n'}
-		}
-
-		// write lines
-		if size, err = file.Write(b.lines[0].data); err != nil {
-			return
-		}
-
-		for _, l := range b.lines[1:] {
-			if _, err = file.Write(eol); err != nil {
-				return
-			}
-			if _, err = file.Write(l.data); err != nil {
-				return
-			}
-			size += len(eol) + len(l.data)
-		}
-		return
-	}
-
-	if err = overwriteFile(name, enc, fwriter, withSudo); err != nil {
-		return
-	}
-
```

Comment:
The below code, including `calcHash()`, is now called even if `err` is not nil. It causes a regression: after a failed save, micro shows an error message but at the same time shows in the statusline that the file is not modified (as if it was successfully saved), since `Modified()` erroneously returns false.

Reply:
You're right...missed that.
It makes it now a bit more complex since we've to bubble the written size out of the async goroutine, because currently I assume that the re-hashing fits best into `saveToFile()`, where the whole `buffer` "finalization" resp. housekeeping is done.

Do you agree?

Reply:
Yes I agree, and from what I can see, this code was in `saveToFile()` originally, I don't know why did you move it from there to `overwrite()`.

Reply:
The intention was to don't bubble up `size` through the functions. It was the wrong decision...